crates/astria-composer/src/searcher/mod.rs

```diff
@@ -216,83 +163,47 @@ impl Searcher {
-                    chain_id: chain_id.into_bytes(),
-                    data,
-                })],
-            };
-            Ok(unsigned_tx.into_signed(&sequencer_key))
-        });
-    }
-
-    fn handle_signed_tx(&mut self, tx: SignedTransaction) {
-        use sequencer_client::SequencerClientExt as _;
-        let client = self.sequencer_client.inner.clone();
-        self.submission_tasks.spawn(async move {
-            let rsp = client
-                .submit_transaction_sync(tx)
-                .await
-                .wrap_err("failed to submit transaction to sequencer")?;
-            if rsp.code.is_err() {
-                bail!(
-                    "submitting transaction to sequencer returned with error code; code: \
-                     `{code}`; log: `{log}`; hash: `{hash}`",
-                    code = rsp.code.value(),
-                    log = rsp.log,
-                    hash = rsp.hash,
-                );
-            }
-            Ok(())
+            vec![seq_action]
         });
     }
 
     /// Starts the searcher and runs it until failure
     pub(super) async fn run(mut self) -> eyre::Result<()> {
         self.spawn_collectors();
+        let mut exectuor_handle = self.spawn_executor();
         let wait_for_collectors = self.wait_for_collectors();
-        let wait_for_seq = self.wait_for_sequencer(5, Duration::from_secs(5), 2.0);
-        match tokio::try_join!(wait_for_collectors, wait_for_seq) {
+        let wait_for_executor = self.wait_for_executor();
+        match tokio::try_join!(wait_for_collectors, wait_for_executor) {
             Ok(((), ())) => {}
             Err(err) => return Err(err).wrap_err("failed to start searcher"),
         }
-        // set initial sequencer nonce
-        let address = Address::from_verification_key(self.sequencer_key.verification_key());
-        let nonce_response = self
-            .sequencer_client
-            .inner
-            .get_latest_nonce(address.0)
-            .await
-            .wrap_err("failed to query sequencer for nonce")?;
-        self.sequencer_nonce
-            .store(nonce_response.nonce, Ordering::Relaxed);
 
         loop {
             select!(
                 // serialize and sign sequencer tx for incoming pending rollup txs
-                Some(rollup_tx) = self.new_transactions.recv() => self.handle_pending_tx(rollup_tx),
+                Some(rollup_tx) = self.new_transactions.recv() => self.bundle_pending_tx(rollup_tx),
 
                 // submit signed sequencer txs to sequencer
                 Some(join_result) = self.conversion_tasks.join_next(), if !self.conversion_tasks.is_empty() => {
                     match join_result {
-                        Ok(Ok(signed_tx)) => self.handle_signed_tx(signed_tx),
-                        Ok(Err(e)) => warn!(error.message = %e, error.cause_chain = ?e, "failed to sign sequencer transaction"),
+                        Ok(actions) => self.executor_tx.send(actions).await?,
```

Comment:
This is a) breaking the loop, but also b) not adding error context. Is that intentional